stdlib/Profile/src/Profile.jl

```diff
@@ -63,6 +63,27 @@ macro profile(ex)
     end
 end
 
+export @profile_tasks
+
+"""
+    @profile_tasks
+
+`@profile_tasks <expression>` runs your expression while taking periodic backtraces of a sample of all live tasks (both running and not running).
```

Comment:
i think ideally this would come with more documentation about the use-case, perhaps even an example (may this should go on a manual page rather than docstring, not sure?)

e.g. perhaps explaining/showing how a program with many tasks for the first half and few tasks for the second half would be expected to appear

e.g. perhaps talking about trying to improve cpu-utilization of some mixed IO and cpu-intensive workload and how this could help

i don't want to make docs too big a barrier to landing this (since they can always be improved in follow-up PRs), but i think we do need to give users a bit of guidance on when/why this might be preferred to `@profile`

Reply:
Agreed.

I think that a short comment in the docstring explaining that this tool may be useful to diagnose lock contention, IO bottlenecks, etc, might be useful.

But we should probably write a proper tutorial in a later PR.